unstoppable-ios-app/domains-manager-ios/Modules/MPC/PurchaseMPCFlow/MPCWalletPasswordValidator.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  MPCWalletPasswordValidator.swift
+//  domains-manager-ios
+//
+//  Created by Oleg Kuplin on 29.05.2024.
+//
+
+import Foundation
+
+protocol MPCWalletPasswordValidator {
+    func validateWalletPassword(_ password: String) -> [MPCWalletPasswordValidationError]
+}
+
+extension MPCWalletPasswordValidator {
+    var minMPCWalletPasswordLength: Int { 12 }
+    var maxMPCWalletPasswordLength: Int { 32 }
+    
+    func validateWalletPassword(_ password: String) -> [MPCWalletPasswordValidationError] {
+        let minLength = minMPCWalletPasswordLength
+        let maxLength = maxMPCWalletPasswordLength
+        let numberRegex = try! NSRegularExpression(pattern: "\\d")
+        let specialCharacterRegex = try! NSRegularExpression(pattern: "[!@#$%^&*()_+\\-\\[\\]{};':\"\\\\|,.<>\\/?]+")
```

Comment:
Please extract to `Constants.swift` and locate near `ETHRegexPattern`

Reply:
This regex rules is specifically to satisfy BE requirements when create password for MPC wallet. 
I'm not sure it needs to be on the global Constants level.